### PR TITLE
Module 1 Project Exercise

### PR DESCRIPTION
**Pull request by Jonesey13**
Creating this PR for review purposes. (In future modules could you please create them and add myself and Stephen as reviewers)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -4,9 +4,29 @@
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
 
-@app.route('/')
+@app.route('/index', methods=['Get', 'Put'])
```
Comment (Jonesey13) [inline app.py]: The root (`/`) route has been replaced by `/index`

This will need to be fixed before this is merged in

File: app.py [hunk 2]
```diff
@@ -4,9 +4,29 @@
...
+    todos = session.get_items()
+    todos = sorted(todos, key=lambda k: k['status'], reverse=True)
+    return render_template('index.html', todos = todos)
+
+@app.route('/index/add', methods=['Post'])
+def add_todo():
+    session.add_item(request.form.get('title'))
+    return redirect('/index')
+
+@app.route('/index/update/<int:todo_id>', methods=['Get','Put'])
```
Comment (Jonesey13) [inline app.py]: In reality we only use GET and POST requests throughout this code

File: templates/index.html [hunk 3]
```diff
@@ -2,17 +2,39 @@
...
+        {% endif %}
+      {% endfor %}
+    </ul>
   </div>
+  <form action="/index/add" method="POST">
+    <div class="input-group mb-3">
+      <div class="input-group-prepend">
+        <button class="btn btn-outline-secondary" type="submit" id="title">Add Item</button>
+      </div>
+      <input type="text" id="title" name="title" class="form-control" placeholder="" aria-label="Example text with button addon" aria-describedby="button-addon1">
```
Comment (Jonesey13) [inline templates/index.html:46]: Maybe add the `required` attribute here (+ add server side validation as well?)

File: templates/index.html [hunk 4]
```diff
@@ -2,17 +2,39 @@
...
+<div class="row">
+  <div class="col-md-12">
+    <h2>Items</h2>
+    <ul class="list-group mb-4">
+      {% for todo in todos %}
+        {% if todo.status == 'Not Started' %}
+          <li class="list-group-item">
+            {{ todo.title }} - {{ todo.status }}
+            <div style="float: right;">
+              <a class="btn btn-outline-danger btn-sm" href="/index/delete/{{ todo.id }}">Delete Item</a>
```
Comment (Jonesey13) [inline templates/index.html]: FYI: You could replace the explicit url with the flask method `url_for()` here

**Review by Jonesey13 — COMMENTED** [review]
Looks good on the whole with one minor issue:
* The url route for the main page has been changed (keeping it unchanged is an implicit requirement of the first exercise)

The rest are just minor comments about styling

**Review by JoeWFisher — APPROVED** [review]

**Pull request merged**